modular_skyrat/modules/deployables/code/deployable.dm

```diff
@@ -0,0 +1,741 @@
+		return
+
+	target.pixel_y = target.base_pixel_y + target.body_position_pixel_y_offset
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/CanAttack(the_target)
+	if(target in buckled_mobs) // Can't kick up! :-)
+		return FALSE
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/death(gibbed)
+	if(power_storage && power_storage >= 5000)
+		explosion(src, -1, (power_storage / 12500), (power_storage / 12500), (power_storage / 10000), -1)
+	QDEL_NULL(particles)
+	return ..()
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/proc/check_smoke()
+	if(!power_storage)
+		return
+	switch(power_storage)
+		if(-INFINITY to 5000)
+
+		if(5000 to 20000)
+			particles = new /particles/smoke/robot/BAPer/mild()
+		if(20000 to 35000)
+			particles = new /particles/smoke/robot/BAPer/bad()
+		if (35000 to 40000)
+			particles = new /particles/smoke/robot/BAPer/full()
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/proc/dir_change(datum/source, olddir, newdir)
+	SIGNAL_HANDLER
+
+	if(!particles)
+		return
+	var/truedir = dir
+	if(newdir && (truedir != newdir))
+		truedir = newdir
+
+	switch(truedir)
+		if(NORTH)
+			particles.position = list(-4, 8, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(0.2, -0.2))
+		if(EAST)
+			particles.position = list(4, 8, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(-0.8, 0.2))
+		if(SOUTH)
+			particles.position = list(4, 0, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(0.2, -0.2))
+		if(WEST)
+			particles.position = list(-4, 0, 0)
+			particles.drift = generator("vector", list(0, 0.4), list(0.8, -0.2))
+
+/particles/smoke/robot/BAPer
+	position = list(4, 8, 0)
+	scale = 0.8
+	icon_state = list("steam_1" = 1, "steam_2" = 1, "steam_3" = 2)
+
+/particles/smoke/robot/BAPer/mild
+	color = "#FFFFFF"
+
+/particles/smoke/robot/BAPer/bad
+	color = "#ababab"
+
+/particles/smoke/robot/BAPer/full
+	color = "#636363"
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(has_gravity())
+		playsound(src, 'modular_skyrat/master_files/sound/effects/robot_step.ogg', 100, TRUE)
+	// I know footstep is a mob variable but its so hardcoded and nonmodular...
+
+/obj/structure/bed/borg_action_pacifier/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change = TRUE)
+	. = ..()
+	if(has_gravity())
+		playsound(src, 'modular_skyrat/master_files/sound/effects/robot_step.ogg', 100, TRUE)
+
+/mob/living/simple_animal/hostile/borg_action_pacifier/emp_act(severity)
+	. = ..()
+	if (. & EMP_PROTECT_SELF)
+		return
+	if(power_storage)
+		power_storage = 0
+	if(particles)
+		QDEL_NULL(particles)
+
+/obj/structure/bed/borg_action_pacifier/emp_act(severity)
+	. = ..()
+	if (. & EMP_PROTECT_SELF)
+		return
+	obj_flags |= EMPED
+	undeploy()
+
+/obj/item/grenade/borg_action_pacifier_grenade/proc/ventilate_effect(turf/open/location, vent_scale)
```

Comment:
Documentation please!

Reply:
Also, there's no other proc that can handle doing all of this for you? Surely there's already a proc that creates smoke at a location...

Reply:
AFAIK this is special modular code that is a fake gas, basically water vapor without its effect. We use it for bongs originally of course :-)